backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -798,22 +844,33 @@ public Project getProjectByID(String id) throws InvalidProtocolBufferException {
   public Project setProjectShortName(String projectId, String projectShortName, UserInfo userInfo)
       throws InvalidProtocolBufferException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      Query query = session.createQuery(GET_PROJECT_BY_SHORT_NAME_AND_OWNER_HQL);
-      query.setParameter("projectShortName", projectShortName);
-      query.setParameter("vertaId", authService.getVertaIdFromUserInfo(userInfo));
-      ProjectEntity projectEntity = (ProjectEntity) query.uniqueResult();
-      if (projectEntity != null) {
-        Status status =
-            Status.newBuilder()
-                .setCode(Code.ALREADY_EXISTS_VALUE)
-                .setMessage("Project already exist with given short name")
-                .build();
-        throw StatusProto.toStatusRuntimeException(status);
+
+      List<String> accessibleProjectIds =
+          roleService.getAccessibleResourceIds(
+              null,
+              new CollaboratorUser(authService, userInfo),
+              ResourceVisibility.PRIVATE,
+              ModelDBServiceResourceTypes.PROJECT,
+              Collections.emptyList());
+
+      if (!accessibleProjectIds.isEmpty()) {
+        Query query = session.createQuery(GET_PROJECT_BY_SHORT_NAME_HQL);
+        query.setParameter("projectShortName", projectShortName);
+        query.setParameterList(ModelDBConstants.PROJECT_IDS, accessibleProjectIds);
```

Comment:
I'm unsure why we need to get all project ids to check name collision. We just need to verify within a workspace. Maybe we should move the collision checking to the collaborator service?

Reply:
i forget to add workspace only check here, will add it.